client/modules/IDE/components/Editor.jsx

```diff
@@ -135,7 +135,25 @@ class Editor extends React.Component {
         this.tidyCode();
       }
     });
-
+    let clicks = 0;
+    this._cm.on('gutterClick', (instance, n, gutter, e) => {
+      if (clicks >= 2)clicks = 0;
+      clicks += 1;
+      if (clicks === 1) {
+        setTimeout(() => {
```

Comment:
is there a way to do this without using a `setTimeout`? this clicks counting just seems... prone to bugs.

Reply:
I can remove clicks counting but not the timeout. As timeout is needed to differentiate between a click and a double click. Still, I'll do some research to find an alternative.

Reply:
I tried other ways to resolve this but ended up with some bugs. But the above one doesn't give me any error. I'll just shorten the timeout.

Reply:
i think it's okay if a click or double click highlights the line. that's the behavior in VSC at least.

Reply:
That's what this is doing. What do you mean exactly? Just highlight and not select?

Reply:
i guess i'm wondering why you need to differentiate between a click and a double click?

Reply:
That's because I wanted to differentiate between folding and highlighting. Single click on the line containing folding marker will highlight the folded code. But I wanted to highlight that specific line. That's it.